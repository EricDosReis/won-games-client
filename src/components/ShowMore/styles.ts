import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    cursor: pointer;
    font-weight: bold;
    padding: ${theme.spacings.medium};
    text-align: center;
    text-transform: uppercase;

    &:hover,
    &:focus {
      svg {
        transform: translateY(0.5rem);
      }
    }

    span {
      display: block;
    }

    svg {
      color: ${theme.colors.primary};
      transition: transform ${theme.transition.fast};
    }
  `}
`;
