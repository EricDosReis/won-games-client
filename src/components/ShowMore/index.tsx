import { KeyboardArrowDown as ArrowDown } from '@styled-icons/material-outlined/KeyboardArrowDown';

import * as S from './styles';

export type ShowMoreProps = {
  onClick?: () => void;
};

const ShowMore = ({ onClick }: ShowMoreProps) => (
  <S.Wrapper role="button" onClick={onClick}>
    <span>Show More</span>

    <ArrowDown size="34" />
  </S.Wrapper>
);

export default ShowMore;
