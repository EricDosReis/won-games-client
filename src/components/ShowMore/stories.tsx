import { Story, Meta } from '@storybook/react/types-6-0';

import ShowMore, { ShowMoreProps } from '.';

export default {
  title: 'ShowMore',
  component: ShowMore,
  argTypes: {
    onClick: { action: 'clicked' },
  },
  parameters: {
    backgrounds: {
      default: 'won-dark',
    },
  },
} as Meta;

export const Basic: Story<ShowMoreProps> = args => <ShowMore {...args} />;
