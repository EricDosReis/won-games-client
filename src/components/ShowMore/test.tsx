import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import ShowMore from '.';

describe('<ShowMore />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<ShowMore />);

    expect(screen.getByText(/show more/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
