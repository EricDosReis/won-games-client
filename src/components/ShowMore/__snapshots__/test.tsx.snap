// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowMore /> should render correctly 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  color: #FAFAFA;
  cursor: pointer;
  font-weight: bold;
  padding: 3.2rem;
  text-align: center;
  text-transform: uppercase;
}

.c0:hover svg,
.c0:focus svg {
  -webkit-transform: translateY(0.5rem);
  -ms-transform: translateY(0.5rem);
  transform: translateY(0.5rem);
}

.c0 span {
  display: block;
}

.c0 svg {
  color: #F231A5;
  -webkit-transition: -webkit-transform 0.1s ease-in-out;
  -webkit-transition: transform 0.1s ease-in-out;
  transition: transform 0.1s ease-in-out;
}

<div
  class="c0"
  role="button"
>
  <span>
    Show More
  </span>
  <svg
    aria-hidden="true"
    class="c1"
    fill="currentColor"
    focusable="false"
    height="34"
    viewBox="0 0 24 24"
    width="34"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
    />
  </svg>
</div>
`;
